File: 5 - Form Template Method/C#/MPG.FormTemplateMethod.After/CapitalStrategyAdvisedLine.cs
﻿namespace MPG.FormTemplateMethod.After
{
    public class CapitalStrategyAdvisedLine : CapitalStrategy
    {
        public override double Capital(Loan loan)
        {
            // TODO - riskAmountFor * duration * riskFactor is common in CapitalStrategyAdvisedLine & CapitalStrategyTermLoan
            return RiskAmountFor(loan) * Duration(loan) * RiskFactor(loan);
        }

        protected override double RiskAmountFor(Loan loan)
        {
            return loan.GetCommitment() * loan.GetUnusedPercentage();
        }
    }
}

File: 5 - Form Template Method/Python/after/capital_strategy_advised_line.py
from after.capital_strategy import CapitalStrategy


class CapitalStrategyAdvisedLine(CapitalStrategy):
    def __init__(self):
        CapitalStrategy.__init__(self)

    def risk_amount_for(self, loan):
        return loan.get_commitment() * loan.get_unused_percentage()
